lms/djangoapps/certificates/tests/test_webview_views.py

```diff
@@ -947,6 +968,55 @@ def test_html_view_certificate_available_date_for_instructor_paced_courses(self,
         )
         self.assertContains(response, date)
 
+    @override_settings(FEATURES=FEATURES_WITH_CERTS_ENABLED)
+    @ddt.data(
+        (True, False),
+        (False, False),
+        (True, True),
+        (False, True)
+    )
+    @ddt.unpack
+    def test_html_view_certificate_display_date(self, self_paced, date_override):
+        """
+        Test certificate web view should display the correct date on the
+        certificate in all cases:
+            * self-paced, no date override
+            * instructor-paced with certificate_available_date
+            * self-paced with date override
+            * instructor-paced with date override
+        """
+        self.course.self_paced = self_paced
+        if date_override:
+            self._add_certificate_date_override(datetime.datetime(2021, 5, 11))
+            self.cert.save()
+        today = datetime.datetime.utcnow()
+        self.course.certificate_available_date = today + datetime.timedelta(-2)
+        self.store.update_item(self.course, self.user.id)
+        self._add_course_certificates(count=1, signatory_count=1, is_active=True)
+        test_url = get_certificate_url(
+            user_id=self.user.id,
+            course_id=str(self.course.id),
+            uuid=self.cert.verify_uuid
+        )
+
+        with override_waffle_switch(AUTO_CERTIFICATE_GENERATION_SWITCH, active=True):
+            response = self.client.get(test_url)
+
+        if date_override:
+            expected_date = self.cert.date_override.date
```

Comment:
If I put a breakpoint after this line and check `expected_date`, it is indeed the value that I set on line `990`. `self.cert` has a `date_override`, as it should. 

But the response (you can check with `response.content` and `command+F` for "Issued On", then look a little to the right and see the date displayed) will show one of the other dates, depending which test you’re in.